ste/sourceInfoProvider-Local_linux.go

```diff
@@ -0,0 +1,189 @@
+// +build linux
+
+package ste
+
+import (
+	"errors"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"golang.org/x/sys/unix"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func (f localFileSourceInfoProvider) HasUNIXProperties() bool {
+	return true
+}
+
+func (f localFileSourceInfoProvider) GetUNIXProperties() (common.UnixStatAdapter, error) {
+	// Can we use statx?
+	var uname unix.Utsname
+	err := unix.Uname(&uname)
+	if err != nil {
+		return nil, err
+	}
+
+	linuxVersion := regexp.MustCompile("(\\d+\\.\\d+\\.\\d+)").FindString(string(uname.Release[:]))
+	splits := strings.Split(linuxVersion, ".")
+
+	if len(splits) != 3 {
+		return nil, errors.New("Failed to parse linux version " + linuxVersion)
+	}
+
+	versions := make([]int64, 3)
+	for k, v := range splits {
+		versions[k], _ = strconv.ParseInt(v, 10, 64)
+	}
+
+	var resp common.UnixStatAdapter
+
+	if versions[0] > 4 || (versions[0] == 4 && versions[1] > 11) { // Can we statx?
+		var stat unix.Statx_t
+		// dirfd is a null pointer, because we should only ever be passing relative paths here, and directories will be passed via transferInfo.Source.
+		// AT_SYMLINK_NOFOLLOW is not used, because we automagically resolve symlinks. TODO: Add option to not follow symlinks, and use AT_SYMLINK_NOFOLLOW when resolving is disabled.
+		err = unix.Statx(0, f.transferInfo.Source,
+			unix.AT_STATX_FORCE_SYNC, // We want to sync attributes to ensure correctness.
```

Comment:
we should not set force sync, else it'll cause lot of getattr calls on the wire for NFS/SMB mounted shares f.e.
Note that in NFS/SMB we have the READDIRPLUS/QUERYDIR calls which get file attributes as part of the dir enumeration response, so that a stat call made within the attribute cache expiry time (which is the common case) will be served from the cache. This hugely saves  on the calls made to the server.

Local filesystems mostly ignore this setting as it doesn't make sense, their attributes are strongly consistent at all times.